src/librustdoc/clean/inline.rs

```diff
@@ -217,7 +217,11 @@ fn build_external_function(cx: &DocContext<'_>, did: DefId) -> clean::Function {
     } else {
         hir::Constness::NotConst
     };
-
+    let asyncness =  if cx.tcx.is_async_fn(did) {
```

Comment:
Nit: it might be nicer to have the query return `hir::IsAsync` directly? e.g., `cx.tcx.asyncness(did)` -- I think almost everything else in this PR would be the same.